Lesion inpainting (LIT) integration (#803)
Deep-MI/FastSurfer: doc/overview/modules/LIT.md

# Lesion Inpainting Tool (LIT)

With the **Lesion Inpainting Tool (LIT)** extension, FastSurfer is able to process T1-weighted
images with lesions, such as tumors, cavities, or abnormalities. Since the deep learning-based
tool LIT is designed to paint healthy-looking tissue into those lesions, downstream analysis with
FastSurfer can produce more reliable whole-brain segmentation and cortical surface reconstruction
in cases with significant structural alterations.

> **Note:** The FastSurfer LIT extension is currently experimental. Review the LIT-modified
> outputs before using them for downstream analyses.

## FastSurfer Usage

FastSurfer runs LIT when a lesion mask is passed with `--lesion_mask <path to file>`:

```bash
./run_fastsurfer.sh --t1 /path/to/T1.nii.gz \
                    --lesion_mask /path/to/lesion_mask.nii.gz \
                    --sid subject_id --sd /path/to/output_dir \
                    --fs_license /path/to/license.txt
```

> 

Reviewer: ```suggestion

Lesion inpainting is not compatible with separate processing of the segmentation and surfaces with `--seg_only` and `--surf_only`. If you want to run the surface pipeline, avoid `--seg_only`!
```

> With `--lesion_mask`, FastSurfer:

Reviewer: ```suggestion
With `--lesion_mask <path to file>`, FastSurfer:
```

# Lesion Inpainting Tool (LIT)

> The **Lesion Inpainting Tool (LIT)** is a deep learning-based tool designed to inpaint lesions (such as tumors, cavities, or abnormalities) in T1-weighted MRI images. This allows downstream analysis tools like FastSurfer to perform more accurate whole-brain segmentation and cortical surface reconstruction in cases with significant structural alterations.

Reviewer: ```suggestion
With the **Lesion Inpainting Tool (LIT)** extension, FastSurfer is able to process T1-weighted images with lesions (such as tumors, cavities, or abnormalities). Since the deep learning-based tool LIT is designed to paint healthy-looking tissue into those lesions (such as tumors, cavities, or abnormalities), downstream analysis with FastSurfer results in more accurate whole-brain segmentation and cortical surface reconstruction in cases with significant structural alterations.
```

Author: Changed (currently in local changes, but I'm resolving to keep an overview of unresolved issues)

> **Note:** The FastSurfer LIT integration is currently experimental. Review the LIT-modified
> outputs before using them for downstream analyses.

## Key Features

- **Shape-Independent Inpainting**: Effectively inpaints lesions regardless of their shape or appearance.
- **FastSurfer Integration**: Can be run as part of the main FastSurfer pipeline using the `--lesion_mask` flag.
- **Standalone Mode**: Can be used independently to create inpainted images for other tools.

> - **Postprocessing Tools**: Includes scripts to map lesion masks back into FastSurfer/FreeSurfer outputs and generate anatomy reports.

Reviewer: I would like to change the way this is written. It should be from the perspective of FastSurfer, this currently reads more like the neuro-lit documentation.

So shape-independent inpainting is fine, but standalone is not relevant.

Also, more details are needed for the lesion mask file:
1. image size
2. vox2ras
3. data type (I assume int?)
4. values and their meaning
5. what effect does it have, etc.

Author: I removed this paragraph and steamlined the description. For the lesion mask there aren't really many requirements - any volume that correctly specifies the lesion should work. The lesion mask is just nonzero voxels.

```bash
./run_fastsurfer.sh --t1 /path/to/T1.nii.gz \
                    --lesion_mask /path/to/lesion_mask.nii.gz \
                    --sid subject_id --sd /path/to/output_dir \
                    --fs_license /path/to/license.txt
```

When this flag is provided, FastSurfer will:
1. Inpaint the lesion region using the LIT model.
2. Run the segmentation and surface pipelines on the inpainted image.
3. Automatically map the lesion mask into the final segmentation and surface outputs.

## Postprocessing and Anatomy Reports

LIT provides comprehensive postprocessing tools to integrate lesions into FastSurfer outputs. This includes:
- **Lesion Mapping**: Mapping the lesion mask into the relevant FastSurfer segmentations and stats files.
- **Anatomy Reports**: Identifying structures that are replaced, reduced, or adjacent to the lesion.
- **Surface Masking**: Projecting the lesion onto cortical surfaces.

The current integration updates the main FastSurfer output files in place and preserves the pre-lesion versions either as `.lit` backups or, for some surface-derived files, in the original `.mapped.*` files. The most important lesion-specific outputs are:

> 

Reviewer: I think I would add a full documentation of the outputs here instead of OUTPUT_FILES.md

Author: see below

- `mri/inpainted.lit.nii.gz`, `mri/mask.lit.nii.gz`, and `mri/orig/mask.lit.nii.gz` as the inpainting inputs stored in the subject directory
- `mri/aparc.DKTatlas+aseg.deep.mgz` with backup `mri/aparc.DKTatlas+aseg.deep.lit.mgz`
- `stats/aseg+DKT.VINN.stats` with backup `stats/aseg+DKT.VINN.lit.stats`
- `stats/aparc.DKTatlas+aseg.lesion_report.txt`
- `stats/aseg.lesion_report.txt`

> - `stats/lesion_impact_summary.yaml`

Reviewer: This is probably difficult change right now, but I think this is the only file in YAML format in FastSurfer outputs.
FastSurferCC already introduced the only JSON.
I would really like file formats to be more standardized and consistent; would it be able to support other file formats?

Author: I agree with you. Let's change it in the next FastSurfer release? Since LIT is experimental this should be fine - I'll keep it in mind
